app/src/main/java/ch/epfl/cs311/wanderwave/model/remote/BeaconConnection.kt

```diff
@@ -77,53 +77,49 @@ class BeaconConnection(
+                      }
+                      // map the list of flow of Result<ProfileTrackAssociation> to a flow of
+                      // Result<List<ProfileTrackAssociation>>
+                      ?.mapNotNull { flow -> flow.mapNotNull { result -> result.getOrNull() } }
+                      // reduce the list of flow to a single flow that contains the list of
+                      // ProfileTrackAssociation
+                      ?.fold(flowOf(Result.success(listOf<ProfileTrackAssociation>()))) { acc, track
+                        ->
+                        acc.combine(track) { accTracks, track ->
+                          accTracks.map { tracks -> tracks + track }
                         }
-                        // map the list of flow of Result<ProfileTrackAssociation> to a flow of
-                        // Result<List<ProfileTrackAssociation>>
-                        ?.mapNotNull { flow -> flow.mapNotNull { result -> result.getOrNull() } }
-                        // reduce the list of flow to a single flow that contains the list of
-                        // ProfileTrackAssociation
-                        ?.fold(flowOf(Result.success(listOf<ProfileTrackAssociation>()))) {
-                            acc,
-                            track ->
-                          acc.combine(track) { accTracks, track ->
-                            accTracks.map { tracks -> tracks + track }
-                          }
-                        } ?: flowOf(Result.failure(Exception("Could not retrieve chosenSongs")))
-
-                // Update the beacon with the profile and track
-                profileAndTracks.collect { result ->
-                  result.onSuccess { profileAndTracks ->
-                    val updatedBeacon = beacon.copy(profileAndTrack = profileAndTracks)
-                    trySend(Result.success(updatedBeacon))
-                  }
-                  result.onFailure { exception ->
-                    trySend(Result.success(beacon))
-                    Log.e("Firestore", "Error getting profile and track: ", exception)
-                  }
+                      } ?: flowOf(Result.failure(Exception("Could not retrieve chosenSongs")))
+
+              // Update the beacon with the profile and track
+              profileAndTracks.collect { result ->
+                result.onSuccess { profileAndTracks ->
+                  val updatedBeacon = beacon.copy(profileAndTrack = profileAndTracks)
+                  trySend(Result.success(updatedBeacon))
+                }
+                result.onFailure { exception ->
+                  trySend(Result.success(beacon))
+                  Log.e("Firestore", "Error getting profile and track: ", exception)
                 }
               }
-            } else {
-              trySend(Result.success(beacon))
-              Log.e("Firestore", "Tracks are not in the correct format ")

```

Comment:
Looks like a lot of formatting stuff here, just to check is this purely formatting or did I miss an actual change past the `coroutineScope.launch`?

Reply:
There is no actual change it's just that the document cannot be null at this state so I just put a !! instead of this ?.let.